classes/abstracts/ActionScheduler_Abstract_QueueRunner.php

```diff
@@ -44,23 +44,25 @@ public function __construct( ActionScheduler_Store $store = null, ActionSchedule
 	 * Process an individual action.
 	 *
 	 * @param int $action_id The action ID to process.
-	 * @param string $context Optional identifer for the context in which this action is being processed, e.g. 'WP CLI' or 'WP Cron'
+	 * @param string $context Optional identifier for the context in which this action is being processed, e.g. 'WP CLI' or 'WP Cron'
 	 *        Generally, this should be capitalised and not localised as it's a proper noun.
 	 */
 	public function process_action( $action_id, $context = '' ) {
 		// Temporarily override the error handler while we process the current action.
 		set_error_handler(
 			/**
-			 * Temporary error handler which can catch errors and convert them into exceptions. This faciliates more
+			 * Temporary error handler which can catch errors and convert them into exceptions. This facilitates more
 			 * robust error handling across all supported PHP versions.
 			 *
 			 * @throws Exception
 			 *
 			 * @param int    $type    Error level expressed as an integer.
 			 * @param string $message Error message.
+			 * @param string $file    File in which the error occurred.
+			 * @param int    $line    Line number at which the error occurred.
 			 */
-			function ( $type, $message ) {
-				throw new Exception( $message );
+			function ( $type, $message, $file, $line ) {
+				throw new ErrorException( $message, 0, $type, $file, $line );
```

Comment:
@james-allan I'm not sure this is enough (or at least, may not work as expected with recent PHP runtimes). When I test:

- `action_scheduler_begin_execute` is fired from [within a try {} block]((https://github.com/woocommerce/action-scheduler/blob/3.8.0/classes/abstracts/ActionScheduler_Abstract_QueueRunner.php#L84)), and the test snippet then triggers an error.
- We enter this error handler, and throw the `ErrorException`.
- We now return to the [matching catch {} block](https://github.com/woocommerce/action-scheduler/blob/3.8.0/classes/abstracts/ActionScheduler_Abstract_QueueRunner.php#L91-L95) which converts the throwable (in this case, the `ErrorException`) into a regular exception.

So, in my own local testing, I do not see the same result you document in the testing instructions.

Reply:
Fwiw, I tested with PHP 8.3.